### ISSUE TITLE
10.11 Loading Modules from a Remote Machine Using Import hooks

### ISSUE THREAD

**Issue by cmbowyer13** [dabeaz/python-cookbook#18]
Hello, I have set up the urlimport.py script as described, but from the jupyternotebook a method or attribute from urlimport.py is not being recognized as such
<img width="858" alt="screen shot 2018-10-05 at 10 22 09 pm" src="https://user-images.githubusercontent.com/28736542/46566493-65787e80-c8ed-11e8-9013-ed57147ce3b0.png">

**Comment by cmbowyer13** [comment]
And the def install_meta() ... is in the file in the same directory.

**Comment by dabeaz** [comment]
Are you sure it's loading the right file?   What does the  `__file__` attribute of the `urlimport` module say?

**Comment by cmbowyer13** [comment]
printing out the__file__ attribute does  return a valid file string. When I tried running different parts from a terminal it worked. But shouldn't it be able to run in a notebook as well?

**Comment by dabeaz** [comment]
Yes, but is it the right file?   In the big picture, I don’t really know on the notebook. It does a lot of things with the environment and I never use it. So I don’t have any specific advice to offer.

Cheers
Dave

**Comment by cmbowyer13** [comment]
Thank you for your time @dabeaz . I have enjoyed learning advanced Python from your books. Do you have any favorite resources for learning more about classes for Python3 in addition to your books? Or anything related to a RasberryPi that you have found to be useful? 

Best,
Caleb
